sympy/solvers/solveset.py

```diff
@@ -3445,6 +3445,9 @@ def _separate_poly_nonpoly(system, symbols):
     denominators = set()
     poly = None
     for eq in system:
+        # Convert equality to expression
```

Comment:
This has to come after the denominators are identified, otherwise two terms with denominators might cancel, e.g. `Eq(2*x + 1/x, x + 1/x)` does not have a solution of 0.  (And that can be added as a test.)